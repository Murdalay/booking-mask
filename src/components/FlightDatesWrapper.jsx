import React from 'react';
import styled from 'styled-components';

const FlightDatesWrapper = styled.div`
  height: ${props => props.theme.sizes.controlHeight}px;
  min-width: ${props => props.theme.sizes.inputMinWidth + 160}px;
  width: 100%;
  display: table-cell;
  bacground-color: white;
  input {
    height: 100%;
    font-size: 1.5rem;
    padding-left: 10px;
  }
  .DateRangePicker {
    height: 100%;
    width: 100%;
    & > div {
      height: 100%;
      width: 100%;
    }
  }
  .DateInput {
    height: 100%;
    width: 50%;
    &:last-child {
      input {
        padding-left: 35px;
      }
    }
  }
  .DateRangePickerInput_arrow {
    position: absolute;
    margin-left: -23px;
    top: 25%;
  }
  .DateRangePickerInput {
    height: 100%;
    width: 100%;
  }
`;

export default React.memo(FlightDatesWrapper);
