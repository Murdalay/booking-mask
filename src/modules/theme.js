export const rowHeight = 70;
export const controlHeight = 55;
export const appColor = '#00b4d5';
export const secondaryColor = '#88e4f5';
export const actionColor = '#66b4e5';
export const inputMinWidth = 150;
export const defaultIconColor = '#444';
export const defaultTextColor = '#555';
export const borderColor = 'rgb(204, 204, 204)';
export const easing = 'cubic-bezier(0.35, 0.01, 0.77, 0.34);';
export const inputFontSize = '1.5rem';
export const inputBackgroundColor = '#fff';

export default {
  breakpoints: [400, 768, 1024, 1280, 1920],
  palette: {
    primary: appColor,
    secondary: secondaryColor,
    actionColor,
    appColor,
    borderColor,
    defaultTextColor,
    defaultIconColor,
    inputBackgroundColor,
  },
  sizes: {
    rowHeight,
    controlHeight,
    borderColor,
    inputMinWidth,
    inputFontSize,
  },
  input: {
    borderRadius: {
      xs: 0,
      lg: 0,
      xl: 0,
      default: 0,
    },
  },
  button: {
    borderRadius: {
      xs: 0,
      lg: 0,
      xl: 0,
    },
    padding: {
      lg: [12, 28],
      xl: [14, 32],
    },
  },
};
