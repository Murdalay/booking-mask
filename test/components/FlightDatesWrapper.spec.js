import React from 'react';
import theme from 'modules/theme';

import FlightDatesWrapper from 'components/FlightDatesWrapper';

describe('FlightDatesWrapper', () => {
  const wrapper = shallow(<FlightDatesWrapper theme={theme} />);

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
