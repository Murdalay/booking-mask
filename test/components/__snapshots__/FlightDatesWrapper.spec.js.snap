// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightDatesWrapper should render properly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "FlightDatesWrapper-aij2n6-0",
        "isStatic": false,
        "rules": Array [
          "height:",
          [Function],
          "px;min-width:",
          [Function],
          "px;width:100%;display:table-cell;bacground-color:white;input{height:100%;font-size:1.5rem;padding-left:10px;}.DateRangePicker{height:100%;width:100%;& > div{height:100%;width:100%;}}.DateInput{height:100%;width:50%;&:last-child{input{padding-left:35px;}}}.DateRangePickerInput_arrow{position:absolute;margin-left:-23px;top:25%;}.DateRangePickerInput{height:100%;width:100%;}",
        ],
      },
      "displayName": "FlightDatesWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "FlightDatesWrapper-aij2n6-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "breakpoints": Array [
        400,
        768,
        1024,
        1280,
        1920,
      ],
      "button": Object {
        "borderRadius": Object {
          "lg": 0,
          "xl": 0,
          "xs": 0,
        },
        "padding": Object {
          "lg": Array [
            12,
            28,
          ],
          "xl": Array [
            14,
            32,
          ],
        },
      },
      "input": Object {
        "borderRadius": Object {
          "default": 0,
          "lg": 0,
          "xl": 0,
          "xs": 0,
        },
      },
      "palette": Object {
        "actionColor": "#66b4e5",
        "appColor": "#00b4d5",
        "borderColor": "rgb(204, 204, 204)",
        "defaultIconColor": "#444",
        "defaultTextColor": "#555",
        "inputBackgroundColor": "#fff",
        "primary": "#00b4d5",
        "secondary": "#88e4f5",
      },
      "sizes": Object {
        "borderColor": "rgb(204, 204, 204)",
        "controlHeight": 55,
        "inputFontSize": "1.5rem",
        "inputMinWidth": 150,
        "rowHeight": 70,
      },
    }
  }
/>
`;
